src/mame/drivers/hp9845.cpp

```diff
@@ -3748,8 +3635,15 @@ MACHINE_CONFIG_START(hp9845_base_state::hp9845_base)
 	m_ppu->set_9845_boot_mode(true);
 	m_ppu->set_rw_cycles(6 , 6);
 	m_ppu->set_relative_mode(true);
-	m_ppu->set_irq_acknowledge_callback(FUNC(hp9845_base_state::irq_callback));
-	m_ppu->pa_changed_cb().set(FUNC(hp9845_base_state::pa_w));
+	m_ppu->set_irq_acknowledge_callback("io_sys" , FUNC(hp98x5_io_sys_device::irq_callback));
+	m_ppu->pa_changed_cb().set(m_io_sys , FUNC(hp98x5_io_sys_device::pa_w));
+
+	HP98X5_IO_SYS(config , m_io_sys , 0);
+	m_io_sys->irl().set([this](int state) { m_ppu->set_input_line(HPHYBRID_IRL , state); });
+	m_io_sys->irh().set([this](int state) { m_ppu->set_input_line(HPHYBRID_IRH , state); });
```

Comment:
You can do this without the trampolines as:
```
 	m_io_sys->irl().set_inputline(m_ppu, HPHYBRID_IRL);
 	m_io_sys->irh().set_inputline(m_ppu, HPHYBRID_IRH);
```
We consider `device_execute_interface` input lines to be important enough to have special support.